medical_prescription_sale_stock_us/models/medical_prescription_order_line.py

```diff
@@ -61,43 +64,54 @@ def _compute_dispense_remain(self):
                 )
 
             if not len(procurement_ids):
+                rec_id.dispense_remain_qty, rec_id.dispense_remain_day = 0, 0
                 continue
 
-            proc_id = procurement_ids[-1]
-
-            if rec_id.dispense_uom_id != proc_id.product_uom:
-                dispense_qty = proc_id.product_uom._compute_qty_obj(
-                    proc_id.product_qty, rec_id.dispense_uom_id
-                )
-            else:
-                dispense_qty = proc_id.product_qty
+            first_proc = procurement_ids[0]
+            date_first_dispense = fields.Datetime.from_string(
+                first_proc.date_planned,
+            )
+            date_delta = datetime.now() - date_first_dispense
+            days_passed = date_delta.days
 
             if day_uom_id != rec_id.duration_uom_id:
-                day_qty = rec_id.duration_uom_id._compute_qty_obj(
-                    rec_id.duration, day_uom_id,
+                day_qty = self.env['product.uom']._compute_qty_obj(
+                    rec_id.duration_uom_id, rec_id.duration, day_uom_id,
                 )
             else:
                 day_qty = rec_id.duration
-
-            date_dispense = fields.Datetime.from_string(proc_id.date_planned)
-            daily_qty = rec_id.total_allowed_qty / float(day_qty)
-            days_dispensed = dispense_qty / daily_qty
-            date_delta = datetime.now() - date_dispense
-            days_remain = days_dispensed - date_delta.days
-
-            rec_id.last_dispense_remain_day = days_remain
-            rec_id.last_dispense_remain_qty = days_remain * daily_qty
-            rec_id.last_dispense_remain_percent = 100.0 * (
-                days_remain / days_dispensed
-            )
+            total_qty = rec_id.qty * (rec_id.refill_qty_original + 1.0)
+            daily_qty = total_qty / float(day_qty)
+            estimated_use = days_passed * daily_qty
+
+            total_dispensed = 0
+            for proc_id in procurement_ids:
+                if rec_id.dispense_uom_id != proc_id.product_uom:
+                    dispense_qty = self.env['product.uom']._compute_qty_obj(
+                        proc_id.product_uom,
+                        proc_id.product_qty,
+                        rec_id.dispense_uom_id,
+                    )
+                else:
+                    dispense_qty = proc_id.product_qty
+                total_dispensed += dispense_qty
+
+            remaining_units = total_dispensed - estimated_use
+            remaining_units = 0 if remaining_units < 0 else remaining_units
+            rec_id.dispense_remain_qty = remaining_units
+            rec_id.dispense_remain_day = remaining_units / daily_qty
 
     @api.multi
-    @api.depends('qty',
-                 'dispensed_qty',
-                 'exception_dispense_qty',
-                 'pending_dispense_qty',
-                 'refill_qty_original',
-                 )
+    @api.depends(
```

Comment:
If the depends are needed here, we'll need to move into https://github.com/laslabs/vertical-medical/blob/release/9.0/medical_prescription_sale_stock_us/models/medical_prescription_order_line.py#L101

The attrs are stored if I'm remembering correctly?